### PR TITLE
Integrate announcement database functionality

### PR DESCRIPTION
**Pull request by dakriy**
Also has some user interaction with announcements, also create some basic unit tests for the announcements.

### CODE REVIEW

File: Legion.BackEnd/app/Announcement.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    protected $fillable = [
+        'title', 'content'
+    ];
+
+    /**
+     * The table associated with the model.
+     *
+     * @var string
+     */
+    protected $table = 'announcements';
```
Comment (blakestone95) [inline Legion.BackEnd/app/Announcement.php]: You really don't have to comment this.  I think it's pretty obvious what this does lol.  If it's best practices, though, go ahead.

Comment (dakriy) [reply Legion.BackEnd/app/Announcement.php]: Might as well have it for consistency is my reasoning.

File: Legion.BackEnd/database/migrations/2018_07_17_021708_create_announcements_table.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('announcements', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('title');
+            // Will be able to store up to 64KiB a field with play TEXT field. Should be good enough.
+            // Can go up to 4 GiB if needed with LONGTEXT...
```
Comment (blakestone95) [inline Legion.BackEnd/database/migrations/2018_07_17_021708_create_announcements_table.php]: I don't think we'll ever exceed 64KiB lol

File: Legion.BackEnd/database/migrations/2018_07_17_021708_create_announcements_table.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            $table->text('content');
+            $table->unsignedInteger('user_id')->index();
+            $table->unsignedInteger('last_modified_by')->index();
+            $table->timestamps();
+            // Enable announcement soft deleting for records.
+            $table->softDeletes();
+        });
+    }
```
Comment (blakestone95) [inline Legion.BackEnd/database/migrations/2018_07_17_021708_create_announcements_table.php]: Is there a creation date?  I only see last modified?  Also, is there a created by?  Only need to store user IDs, don't worry about names (you may not have, I can't really tell).

File: Legion.BackEnd/database/migrations/2018_07_17_021708_create_announcements_table.php (hunk 2 above)
Comment (dakriy) [reply Legion.BackEnd/database/migrations/2018_07_17_021708_create_announcements_table.php]: The one below it is like 255 chars so, meh.

File: Legion.BackEnd/database/migrations/2018_07_17_021708_create_announcements_table.php (hunk 3 above)
Comment (dakriy) [reply Legion.BackEnd/database/migrations/2018_07_17_021708_create_announcements_table.php]: I store the user id, and there is a creation and last modified dates, those are packaged in the timestamps() function right above the softDeletes.

File: Legion.BackEnd/app/Announcement.php (hunk 1 above)
Comment (blakestone95) [reply Legion.BackEnd/app/Announcement.php]: Alright, at long as it doesn't get out of hand.

File: Legion.BackEnd/database/migrations/2018_07_17_021708_create_announcements_table.php (hunk 3 above)
Comment (blakestone95) [reply Legion.BackEnd/database/migrations/2018_07_17_021708_create_announcements_table.php]: What if someone other than the create modified it, though?

Comment (dakriy) [reply Legion.BackEnd/database/migrations/2018_07_17_021708_create_announcements_table.php]: The time still gets updated. I just update the modified_by field or whatever I called it.

File: Legion.BackEnd/database/migrations/2018_07_17_021708_create_announcements_table.php (hunk 2 above)
Comment (blakestone95) [reply Legion.BackEnd/database/migrations/2018_07_17_021708_create_announcements_table.php]: That's fine.  Do you know if it uses up 64KiB per record, or if it's smart and only uses what it needs?

Comment (dakriy) [reply Legion.BackEnd/database/migrations/2018_07_17_021708_create_announcements_table.php]: It uses what it needs.

Comment (dakriy) [reply Legion.BackEnd/database/migrations/2018_07_17_021708_create_announcements_table.php]: https://stackoverflow.com/questions/37518598/space-occupied-by-text-type-column-in-mysql

**Pull request merged**